ttnn/cpp/ttnn/operations/experimental/ccl/all_reduce/device/all_reduce_op.cpp

```diff
@@ -66,20 +67,30 @@ namespace experimental {
 namespace ccl {
 
 static AllReduceStrategy choose_all_reduce_strategy(
-    const Tensor& input_tensor, uint32_t num_devices, uint32_t num_links) {
+    const Tensor& input_tensor, uint32_t num_devices, uint32_t num_links, ttnn::ccl::Topology topology) {
     auto shape = input_tensor.get_logical_shape();
     auto rank = shape.rank();
 
     uint32_t all_reduce_dim = -1;
     bool optimized_version = false;
 
+    if (num_devices == 2) {
+        // 2 devices == n300 == linear topology
```

Comment:
Probably here because of debug but delete this before merging

Reply:
@jvegaTT - can you please clarify your concern here? Is it specifically the comment or the entire conditional block? If the latter, then this change is required for correctness. I kept it intentionally separate from the later 

```
    if (topology == ttnn::ccl::Topology::Linear) {
        // reduce scatter doesn't reliably support line topology yet
        optimized_version = false;
    }
```
because they are applying those modifications for different reasons:

This block in this comment is related to topology. If we are on a 2 device system, it implies line (atleast today - in the future we will need to generalize to make the op infra more thoroughly detect line vs ring be checking ethernet link connectivity or querying fabric).

The second block is temporarily there because reduce scatter isn't stable on line topology, and it's used for the optimized version of all-reduce, so we disable the optimized version if we are on a line.

Reply:
OK, it was just the // 2 devices == n300 == linear topology comment. But on second look I think that it is good to clarify this so let's keep it.